### PR TITLE
3.8.0 release

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

File: completion/git-machete.completion.bash [hunk 1]
```diff
@@ -41,13 +43,16 @@ _git_machete() {
         add) __gitcomp "$common_opts $add_opts" ;;
         advance) __gitcomp "$common_opts $advance_opts" ;;
         anno) __gitcomp "$common_opts $anno_opts" ;;
+        clean) __gitcomp "$common_opts $clean_opts" ;;
         d|diff) __gitcomp "$common_opts $diff_opts" ;;
         delete-unmanaged) __gitcomp "$common_opts $delete_unmanaged_opts" ;;
         discover) __gitcomp "$common_opts $discover_opts" ;;
         fork-point) __gitcomp "$common_opts $fork_point_opts" ;;
-        github)
+        github) # TODO (#393): Add support in zsh and bash completion for options of `git machete github checkout-prs`
```
Comment (PawelLipski) [inline completion/git-machete.completion.bash]: This is done right?

File: ci/snap/README.md [hunk 2]
```diff
@@ -1,3 +1 @@
 Note that this directory is distinct from top-level [snap/](https://github.com/VirtusLab/git-machete/tree/master/snap) directory.
-This directory only contains a script that triggers a dry-run of snap builds,
-and is **not** used when actually building the snap.
```
Comment (PawelLipski) [inline ci/snap/README.md:3]:
```markdown
This directory is used for the scripts that build and upload snaps.
`snap/` directory contains the config of `snapcraft` tool itself.
```

File: ci/checks/ensure-consistent-style-for-fork-point.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+self_dir=$(cd "$(dirname "$0")" &>/dev/null; pwd -P)
+source "$self_dir"/utils.sh
+
+if git grep -we forkpoint -we Forkpoint -- :^ci/checks/ensure-consistent-style-for-fork-point.sh; then
```
Comment (PawelLipski) [inline ci/checks/ensure-consistent-style-for-fork-point.sh]:
```suggestion
if git grep '[fF]orkpoint' -- :^ci/checks/ensure-consistent-style-for-fork-point.sh; then
```

File: git_machete/docs.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
     "add": "Add a branch to the tree of branch dependencies",
     "advance": "Fast-forward merge one of children to the current branch and then slide out this child",
     "anno": "Manage custom annotations",
+    "clean": "Delete untracked and unmanaged branches and also check out user's open GitHub PRs",
```
Comment (PawelLipski) [inline git_machete/docs.py]:
```suggestion
    "clean": "Delete untracked and unmanaged branches and also optionally check out user's open GitHub PRs",
```

File: docs/source/cli.rst [hunk 5]
```diff
@@ -12,7 +12,8 @@ git machete commands and help topics:
 * :ref:`add`              -- Add a branch to the tree of branch dependencies.
 * :ref:`advance`          -- Fast-forward merge one of children to the current branch and then slide out this child
 * :ref:`anno`             -- Manage custom annotations
-* :ref:`delete-unmanaged` -- Delete local branches that are not present in the definition file`
+* :ref:`clean`            -- Delete untracked managed branches with no downstream branch, delete unmanaged branches and also checkout open PRs for the current user associated with the GitHub token
```
Comment (PawelLipski) [inline docs/source/cli.rst]:
```suggestion
* :ref:`clean`            --     Delete untracked and unmanaged branches and also optionally check out user's open GitHub PRs
```

File: docs/source/cli_help/github.rst [hunk 6]
```diff
@@ -12,30 +12,35 @@ where ``<subcommand>`` is one of: ``anno-prs``, ``checkout-prs``, ``create-pr``,
...
 
   Annotates the branches based on their corresponding GitHub PR numbers and authors.
   Any existing annotations are overwritten for the branches that have an opened PR; annotations for the other branches remain untouched.
   Equivalent to ``git machete anno --sync-github-prs``.
 
-``checkout-prs``:
+``checkout-prs [--all | --by=<github-login> | --mine | <PR-number-1> ... <PR-number-N>]``:
 
-  Check out the head branch of the given pull request (specified by number),
+  Check out the head branch of the given pull requests (specified by number),
```
Comment (PawelLipski) [inline docs/source/cli_help/github.rst]:
```suggestion
  Check out the head branch of the given pull requests (specified by numbers or by a flag),
```

File: git_machete/docs.py [hunk 7]
```diff
@@ -320,14 +346,22 @@
           Any existing annotations are overwritten for the branches that have an opened PR; annotations for the other branches remain untouched.
           Equivalent to `git machete anno --sync-github-prs`.
 
-        <b>`checkout-prs <PR-number>`:</b>
+        <b>`checkout-prs [--all | --by=<github-login> | --mine | <PR-number-1> ... <PR-number-N>]`:</b>
 
-          Check out the head branch of the given pull request (specified by number),
+          Check out the head branch of the given pull requests (specified by number),
```
Comment (PawelLipski) [inline git_machete/docs.py]: Same fix as in .rst

File: snap/snapcraft.yaml [hunk 8]
```diff
@@ -9,10 +9,7 @@ description: |
 # This software is unlikely to be used outside of desktop machines.
 # Let's limit the number of target architectures to reduce the build times.
 architectures:
-  # See https://forum.snapcraft.io/t/how-can-i-publish-a-snap-that-can-provide-binaries-for-all-the-supported-architectures/8556/4
-  # on why not just `build-on: [ amd64, i386 ]`.
-  - build-on: amd64
-  - build-on: i386
+  - build-on: [ amd64, i386 ]
```
Comment (PawelLipski) [inline snap/snapcraft.yaml:12]: Hmmm is this change intentional? 🤔 the linked forum thread indicates that the old form is not equivalent to the new form

File: snap/snapcraft.yaml [hunk 9]
```diff
@@ -23,17 +20,15 @@ license: MIT
 parts:
   git-machete:
     plugin: python
-    python-version: python3
```
Comment (PawelLipski) [inline snap/snapcraft.yaml:26]: I suppose that's the default now?

**Review by PawelLipski — CHANGES_REQUESTED** [review]
A bunch of fixes to be applied on a separate PR to develop. The only potentially major thing is `build-on` in snapcraft.yaml

File: completion/git-machete.completion.bash (hunk 1 above)
Comment (amalota) [reply completion/git-machete.completion.bash]: Yes

File: snap/snapcraft.yaml (hunk 9 above)
Comment (amalota) [reply snap/snapcraft.yaml:26]: Yes

File: snap/snapcraft.yaml (hunk 8 above)
Comment (amalota) [reply snap/snapcraft.yaml:12]: Yeah, i read the linked forum but as far as i remember the old settings only kept building for one architecture and not for both, but just to be sure i will test it again for both options.

Comment (amalota) [reply snap/snapcraft.yaml:12]: @PawelLipski I tested it again to make sure and it does not work as before, it only builds `amd64` architecture, not both. https://github.com/VirtusLab/git-machete/pull/489

Comment (amalota) [reply snap/snapcraft.yaml:12]: Here are the logs for 3 settings that i tried: 

1.

  - build-on: amd64
  - build-on: i386
  https://app.circleci.com/pipelines/github/VirtusLab/git-machete/1832/workflows/6fcbabeb-cc96-4e1d-9439-0da366195ca2/jobs/8169
2.
  - build-on: i386
  - build-on: amd64
https://app.circleci.com/pipelines/github/VirtusLab/git-machete/1831/workflows/41501906-deea-4143-b715-6b5c25a6b938/jobs/8166
3.
  - build-on: [ amd64, i386 ]
  https://app.circleci.com/pipelines/github/VirtusLab/git-machete/1833/workflows/a1663a60-9127-464f-a026-8ef05e9a0a55/jobs/8179

Comment (amalota) [reply snap/snapcraft.yaml:12]: You can also check here https://snapcraft.io/git-machete/releases in the `edge` channel

Comment (PawelLipski) [reply snap/snapcraft.yaml:12]: Okay! good spot, now I see, it's indeed a single `..._multi.snap` package published (which I suppose contains binaries for both `amd64` and `i386`) instead of two separate `..._amd64.snap` and `_i386.snap` packages. Actually... support for `i386`  (i.e. 32-bit systems) has a minuscule significance nowadays anyway :)

Comment (PawelLipski) [reply snap/snapcraft.yaml:12]: Let' leave as is then

**Review by PawelLipski — APPROVED** [review]
Let's do a release on Monday! 🚢
